joplin/groups/fixtures/helpers/group_permissions.py

```diff
@@ -0,0 +1,28 @@
+from django.contrib.auth.models import Permission
+
+
+def add_moderator_permissions(group):
+    group.permissions.add(Permission.objects.get(codename="view_extra_panels"))
+    group.permissions.add(Permission.objects.get(codename="add_contact"))
+    group.permissions.add(Permission.objects.get(codename="change_contact"))
+    group.permissions.add(Permission.objects.get(codename="view_contact"))
+    group.permissions.add(Permission.objects.get(codename="view_snippets"))
+    group.permissions.add(Permission.objects.get(codename="view_user"))
+    group.permissions.add(Permission.objects.get(codename="access_admin"))
+    group.permissions.add(Permission.objects.get(codename="add_document"))
+    group.permissions.add(Permission.objects.get(codename="change_document"))
+    group.permissions.add(Permission.objects.get(codename="delete_document"))
+    group.permissions.add(Permission.objects.get(codename="add_image"))
+    group.permissions.add(Permission.objects.get(codename="change_image"))
+    group.permissions.add(Permission.objects.get(codename="delete_image"))
+
+
+def add_editor_permissions(group):
+    group.permissions.add(Permission.objects.get(codename="view_user"))
+    group.permissions.add(Permission.objects.get(codename="access_admin"))
+    group.permissions.add(Permission.objects.get(codename="add_document"))
+    group.permissions.add(Permission.objects.get(codename="change_document"))
+    group.permissions.add(Permission.objects.get(codename="delete_document"))
+    group.permissions.add(Permission.objects.get(codename="add_image"))
+    group.permissions.add(Permission.objects.get(codename="change_image"))
+    group.permissions.add(Permission.objects.get(codename="delete_image"))
```

Comment:
So the Editor and Moderation are groups that already exist. We want to add these permissions to those groups for every environment we run.

Using a migration script is the right call. But I don't think the logic is quite right. I could be wrong, but I don't think we want to add those permissions to pages, I think we want to add those permissions to the editor and moderator groups only. Those aren't permissions that are relevant for pages.

So like, (in pseudocode):
```
editor_group = Group.objects.filter(name="Editor")
editor_group.permissions.add(Permission.objects.get(codename="view_extra_panels"))
editor_group.permissions.add(Permission.objects.get(codename="add_contact"))
...
```

Reply:
I'm not adding it to pages. I can't see a difference between your psuedocode and what I have above.

Reply:
You're right, I misread the 0002_add_pageviewpermissions.py migration file. I thought it was dealing with these group_permissions. I responded to your comment on editor_for_test_env.py